docs/guides/developer/docs/architecture/api/types.md

```diff
@@ -507,3 +507,156 @@ An entry for a playlist as a JSON object. There are different kinds of entries:
+|---------|---------------------------------|-----------------------------------------------------------------------------------|
+| `value` | [`object`](types.md#basic)      | Required. The value to filter the field by.                                       |
+| `type`  | [`enumeration`](types.md#basic) | Query type. Possible values: SEARCH, WILDCARD. Default: SEARCH                    |
+| `must`  | [`boolean`](types.md#basic)     | Whether the value must be in the field or must not be in the field. Default: true |
+
+Example: Get all events with the series "banana", which do not have the word "conference" anywhere in their description.
+
+```json
+{
+  "series_name": {
+    "value": "banana"
+  },
+  "description": {
+    "value": "*conference*",
+    "type": "WILDCARD",
+    "must": false
+  }
+}
+```
+
+Example: Get all events whose start date lies in a specified range, that do not have "Albert Einstein" as a presenter
+and "conference" anywhere in their title.
+
+```json
+{
+  "start_date_range": {
+    "value": "2023-11-30T16:16:47Z/2025-11-30T16:16:47Z"
+  },
+  "presenter": {
+    "value": "Albert Einstein",
+    "must": false
+  },
+  "title": {
+    "value": "*conference*",
+    "type": "WILDCARD",
+    "must": false
+  }
+}
+```
+
+Example: Get all events with a start date older than 2 years from now (where "now" is every time the policy runs)
+that do not have "Albert Einstein" as a presenter, "conference" anywhere in their title and the series "Archive".
+
+```json
+{
+  "start_date_relative": {
+    "value": "-P2Y0M0D",
+    "type": "LESS_THAN"
+  },
+  "presenter": {
+    "value": "Albert Einstein",
+    "must": false
+  },
+  "title": {
+    "value": "*conference*",
+    "type": "WILDCARD",
+    "must": false
+  },
+  "series_name": {
+    "value": "Archive",
+    "must": false
+  }
+}
+```
+
+Example: Get all events with series “08/15 Algebra - 2024” three days after the actual lecture recording happened.
+
+```json
+{
+  "series_name": {
+    "value": "08/15 Algebra - 2024"
+  } ,
+  "start_date_relative": {
+    "value": "-P0Y0M3D",
+    "type": "LESS_THAN"
+  }
+}
+```
+
+#### Types
+
+- SEARCH: The given value must match the value in the field exactly. The default
+- WILDCARD: The given string must match the value in the field exactly, unless wildcard characters are specified.
+  Supported wildcard characters are:
+  - ?, which matches any single character
+  - *, which can match zero or more characters, including an empty one
+  Avoid beginning patterns with * or ?. This can increase the iterations needed to find matching terms and slow search performance.
+  Can only be used for string values, i.e. not for dates.
+- GREATER_THAN: The given value is greater than the value in field.
+  Can only be used for dates.
+- LESS_THAN: The given value is less than the value in field.
+  Can only be used for dates.
```

Comment:
A classic case of basing my field names off of some internal class, without thinking about what makes sense to the eventual user. Sure, I can rename that, but only if you insist (because then I have to think about if I want to rename it like that everywhere in the code as well, and that is annoying).

Reply:
At the end of the day it won't matter since it's in essence a timestamp comparison, I just don't like it.  Not going to insist on changing it.